scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
-        } while (defaultModuleFiles.contains(documentName));
         ModuleId defaultModuleId = defaultModule.moduleId();
-        DocumentId documentId = DocumentId.create(documentName, defaultModuleId);
-        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), documentName);
+        DocumentId documentId = DocumentId.create(generatedDocumentName, defaultModuleId);
+        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), generatedDocumentName);
         defaultModule.modify().addDocument(documentConfig).apply();
-
         project.currentPackage().ballerinaToml().ifPresent(ballerinaToml -> {
             String tomlFileContent = ballerinaToml.tomlDocument().textDocument().toString();
             ballerinaToml.modify().withContent(tomlFileContent + tomlDependencies).apply();
         });
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            String org = pkgManifest.org().value();
+            String name = pkgManifest.name().value();
+            String pluginName = org + FORWARD_SLASH + name;
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors.contains(pluginName)) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<URL> jarUrls = pluginDesc.dependencies().stream().map(dependency -> {
+                        try {
+                            return Path.of(dependency.getPath()).toUri().toURL();
+                        } catch (MalformedURLException ex) {
+                            throw new RuntimeException(ex);
+                        }
+            }).toList();
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(new URL[0]),
+                    this.getClass().getClassLoader());
```

Comment:
Instead of creating a empty array use `new` method reference. This way `toArray` will create the array of the size required, where as passing `new URL[0]` will cause you to create an empty array and `toArray` to create another array to hold the values
```suggestion
            URLClassLoader ucl = URLClassLoader.newInstance(jarUrls.toArray(URL[]::new),
                    this.getClass().getClassLoader());
```